src/App.tsx

```diff
@@ -225,63 +263,157 @@ export default class App extends React.Component<{}, { expandFolderBrowser: bool
-    const breadcrumbs = this.folderBrowser.current?.breadcrumbs;
-    const entryId = Number.parseInt(selectedNode.id, 10);
-    const path = selectedNode.path;
-    const selectedFolder: LfFolder = {
-      entryId,
-      path,
-      displayName: this.getFolderNameText(entryId, path),
-      displayPath: this.getFolderPathTooltip(path)
-    };
-    if (breadcrumbs) {
-      selectedFolder.breadcrumbs = breadcrumbs;
+  onSelectFolder = async () => {
+    if (!this.repoClient) {
+      throw new Error('Repo Client is undefined.');
     }
-    const nodeId = selectedNode.id;
-    const repoId = (await this.repoClient?.getCurrentRepoId());
-    const waUrl = this.loginComponent.current?.account_endpoints?.webClientUrl;
-    this.selectedNodeUrl = getEntryWebAccessUrl(nodeId, repoId!, waUrl!, selectedNode.isContainer);
-    this.selectedFolderDisplayName = selectedFolder?.displayName;
-    this.setState(() => { return { expandFolderBrowser: false, selectedFolder: selectedFolder } });
-  }
+    if (!this.loginComponent.current?.account_endpoints) {
+      throw new Error('LfLoginComponent is not found.');
+    }
+    const selectedNode = this.repositoryBrowser.current?.currentFolder as LfRepoTreeNode;
+    let entryId = Number.parseInt(selectedNode.id, 10);
+    const selectedFolderPath = selectedNode.path;
+    if (selectedNode.entryType === EntryType.Shortcut) {
+      if (selectedNode.targetId)
+      entryId = selectedNode.targetId;
+    }
+    const repoId = (await this.repoClient.getCurrentRepoId());
+    const waUrl = this.loginComponent.current.account_endpoints.webClientUrl;
+    this.setState({
+      lfSelectedFolder: {
+        selectedNodeUrl: getEntryWebAccessUrl(entryId.toString(), repoId, waUrl, selectedNode.isContainer) ?? '',
+        selectedFolderName: this.getFolderNameText(entryId, selectedFolderPath),
+        selectedFolderPath: selectedFolderPath
+      },
+      shouldShowOpen: false,
+      expandFolderBrowser: false,
+      shouldShowSelect: false,
+    });
+  };
+
+  onClickCancelButton = () => {
+    this.setState({
+      expandFolderBrowser: false,
+      shouldShowOpen: false,
+      shouldShowSelect: false
+    });
+  };
+
+  onEntrySelected = (event: any) => {
+    const treeNodesSelected: LfRepoTreeNode[] = event.detail;
+    this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
+    this.setState({
+      shouldShowOpen: this.getShouldShowOpen(),
+      shouldShowSelect: this.getShouldShowSelect(),
+      shouldDisableSelect: this.getShouldDisableSelect(),
+    });
+  };
+  
   folderCancelClick = () => {
-    this.setState(() => { return { expandFolderBrowser: false } });
-  }
+    this.setState({ expandFolderBrowser: false });
+  };
 
   onClickBrowse = async () => {
-    this.setState(() => { return { expandFolderBrowser: true } }, async () => {
-      await this.initializeTreeAsync()
+    this.setState({ expandFolderBrowser: true}, async () => {
```

Comment:
this doesn't look right.
This won't wait until after the initializeTreeAsync to call the setState on line 321 so why have two separate ones